src/main/java/com/google/step/YTLounge/servlets/KeywordSearch.java

```diff
@@ -0,0 +1,105 @@
+package com.google.step.YTLounge.servlets;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.gson.Gson;
+import com.google.step.YTLounge.data.RequestParameter;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/keywordSearch")
+public class KeywordSearch extends HttpServlet {
+  private static final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * @return an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public YouTube getService() throws GeneralSecurityException, IOException {
+    String key = readSecrets();
+    final YouTubeRequestInitializer keyInitializer = new YouTubeRequestInitializer(key);
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, jsonFactory, null)
+        .setApplicationName("YouTube Lounge")
+        .setYouTubeRequestInitializer(keyInitializer)
+        .build();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    String keywords = RequestParameter.getParameter(request, "words", "");
```

Comment:
It seems like words is a string and not a list. If it's  a string why not call it a query or something?